### PR TITLE
refactoring 

### PR DESCRIPTION
**Pull request by tjstebbing**
- refactor type defs for account/invoice into their own files. 
- refactored how event types work so there are now subtypes. 
- Added first pass 'settings' for payout logic, however need to chat with Raffe about refactoring account upsert logic in API as it's inconsistent throughout the webapi -> api -> store layers.

### CODE REVIEW

**Review by qlpqlp — APPROVED** [review]
Much wow. 🔥🚀

File: pkg/api.go [hunk 1]
```diff
@@ -105,6 +109,34 @@ func (a API) GetAccount(foreignID string) (AccountPublic, error) {
...
+	for k, v := range update {
+		switch k {
+		case "PayoutAddress":
+			acc.PayoutAddress = v.(string)
+		case "PayoutThreshold":
+			acc.PayoutThreshold = v.(string)
+		case "PayoutFrequency":
+			acc.PayoutFrequency = v.(string)
+		default:
+			a.bus.Send(SYS_ERR, fmt.Sprintf("Invalid account setting: %s", k))
```
Comment (raffecat) [inline pkg/api.go:128]: This probably also should return an error so the caller knows it failed.

File: pkg/account.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+// to expose to the outside world (i.e. NOT private keys)
+func (a Account) GetPublicInfo() AccountPublic {
+	return AccountPublic{Address: a.Address, ForeignID: a.ForeignID}
+}
+
+type AccountPublic struct {
+	Address         Address `json:"id"`
+	ForeignID       string  `json:"foreign_id"`
+	PayoutAddress   string  `json:"payout_address"`
+	PayoutThreshold string  `json:"payout_threshold"`
```
Comment (raffecat) [inline pkg/account.go:77]: Shouldn't `PayoutThreshold` be a decimal?

**Review by raffecat — APPROVED** [review]
Refactors make sense to me. A couple of comments but happy to merge now.

File: pkg/account.go (hunk 2 above)
Comment (tjstebbing) [reply pkg/account.go:77]: Good catch :D

**Pull request merged**